modules/foreman_auth_source_ldap.py

```diff
@@ -0,0 +1,233 @@
+  ldap_filter:
+    description: Filter to apply to LDAP searches
+    required: false
+  server_url:
+    description: URL of Foreman server
+    required: true
+  username:
+    description: Username accessing Foreman server
+    required: true
+  password:
+    description: Password of user accessing Foreman server
+    required: true
+  validate_certs:
+    aliases: [ verify_ssl ]
+    description: Whether or not to verify the TLS certificates of Foreman server
+    default: true
+    type: bool
+  organizations:
+    description: List of organizations the authentication source should be assigned to
+    type: list
+  locations:
+    description: List of locations the authentication source should be assigned to
+    type: list
+  state:
+    description: State ot the LDAP authentication source
+    default: present
+    choices: ["present", "absent"]
+'''
+
+EXAMPLES = '''
+- name: LDAP Authentication source
+  foreman_auth_source_ldap:
+    name: "Example LDAP"
+    host: "ldap.example.org"
+    server_url: "https://foreman.example.com"
+    locations:
+      - "Uppsala"
+    organizations:
+      - "Sweden"
+    username: "admin"
+    password: "secret"
+    validate_certs: False
+    state: present
+
+- name: LDAP Authentication with automatic registration
+  foreman_auth_source_ldap:
+    name: "Example LDAP"
+    host: "ldap.example.org"
+    onthefly_register: True
+    account: uid=ansible,cn=sysaccounts,cn=etc,dc=example,dc=com
+    account_password: secret
+    base_dn: dc=example,dc=com
+    groups_base: cn=groups,cn=accounts, dc=example,dc=com
+    server_type: free_ipa
+    attr_login: uid
+    attr_firstname: givenName
+    attr_lastname: sn
+    attr_mail: mail
+    attr_photo: jpegPhoto
+    server_url: "https://foreman.example.com"
+    username: "admin"
+    password: "secret"
+    validate_certs: False
+    state: present
+'''
+
+RETURN = ''' # '''
+
+from ansible.module_utils.foreman_helper import (
+    sanitize_entity_dict,
+    ForemanEntityApypieAnsibleModule,
+)
+
+# This is the only true source for names (and conversions thereof)
+name_map = {
+    'name': 'name',
+    'host': 'host',
+    'port': 'port',
+    'account': 'account',
+    'account_password': 'account_password',
+    'base_dn': 'base_dn',
+    'attr_login': 'attr_login',
+    'attr_firstname': 'attr_firstname',
+    'attr_lastname': 'attr_lastname',
+    'attr_mail': 'attr_mail',
+    'attr_photo': 'attr_photo',
+    'onthefly_register': 'onthefly_register',
+    'usergroup_sync': 'usergroup_sync',
+    'tls': 'tls',
+    'groups_base': 'groups_base',
+    'server_type': 'server_type',
+    'ldap_filter': 'ldap_filter',
+    'locations': 'location_ids',
+    'organizations': 'organization_ids',
+}
+
+
+def main():
+    module = ForemanEntityApypieAnsibleModule(
+        argument_spec=dict(
+            name=dict(required=True),
+            host=dict(required=True),
+            port=dict(type='int'),
```

Comment:
You have a default in the docs, I think we should also default it here.

Reply:
Fixed